crnk-gen/crnk-gen-gradle/src/main/java/io/crnk/gen/gradle/internal/RuntimeClassLoaderFactory.java

```diff
@@ -51,7 +51,11 @@ public RuntimeClassLoaderFactory(Project project, GeneratorModule module) {
 	public URLClassLoader createClassLoader(ClassLoader parentClassLoader, boolean isolate) {
 		Set<URL> classURLs = new HashSet<>(); // NOSONAR URL needed by URLClassLoader
 		classURLs.addAll(getProjectLibraryUrls());
-		classURLs.addAll(getPluginUrls());
+		// TODO: The `getPluginUrls` does not work properly, it immediately fails if executed on Java 17.
+		// The culprit is `(URLClassLoader) getClass().getClassLoader()` this piece of code:
+		// `getClass().getClassLoader()` is no longer an instance of `URLClassLoader` and therefore the cast cannot be done.
```

Comment:
This code `(URLClassLoader) getClass().getClassLoader()` throws class cast exception because the class loader returned is not an instance of URLClassLoader anymore.